recommender/engines/rl/training/data_extraction_step/data_extraction_step.py

```diff
@@ -44,26 +44,26 @@ def __init__(self, pipeline_config):
 
     def __call__(self, data=None) -> Tuple[List[Sars], dict]:
         start = time()
-        sarses = self._generate_proper_sarses()
+        sarses = self._generate_sarses()
         end = time()
-        return sarses, {"sarses_generation_duration": end - start}
+        return sarses, {"sarses_generation_duration": f"{end - start}s"}
 
-    def _generate_synthetic(self):
+    def _need_to_generate_sarses(self) -> bool:
```

Comment:
we have to reconsider this logic -> if it so happens that there are enough user actions and recommendations that satisfy this condition, real_sarses_generator could in theory produce unsatisfactory number of SARSes (or even 0, breaking the pipeline entirely!).

i would change it to something along the lines of: 
1. try to generate as much real sarses as you can
2. fill real sarses with synthetic sarses if necessary (up to a certain amount)

But this should be done in another issue

Reply:
Yep, good point. I totally agree. I created a ticket: https://github.com/cyfronet-fid/recommender-system/issues/324